app/static/src/js/headscripts/gtm.js

```diff
@@ -23,6 +23,44 @@ function add_GTM() {
     GTM_Loaded = true;
 }
 
+function push_to_datalayer(event, category_code = null, category_name = null, category_traversal = null, diagnosis_result = null) {
+    window.dataLayer = window.dataLayer || [];
+    window.dataLayer.push({
+        event: event,
+        category_code: category_code,
+        category_name: category_name,
+        category_traversal: category_traversal,
+        diagnosis_result: diagnosis_result,
+    });
+}
+
+function diagnosed(){
+    const path = window.location.pathname;
+
```

Comment:
Might be worth adding comments here as to what is being covered, Something like... // Covers first page after scope diagnosis (inscope). Sometimes this won't show if user is fast tracked to contact page, so covers that too. Same for other sections.